docs/integrations/opentelemetry.mdx

```diff
@@ -0,0 +1,517 @@
+from opentelemetry._logs import set_logger_provider
+from opentelemetry.sdk._logs import LoggerProvider, LoggingHandler
+from opentelemetry.sdk._logs.export import BatchLogRecordProcessor, ConsoleLogExporter
+
+from fastmcp import FastMCP
+from fastmcp.utilities.logging import get_logger
+
+# Configure OpenTelemetry
+resource = Resource(attributes={
+    "service.name": "my-fastmcp-server",
+    "service.version": "1.0.0",
+})
+
+# Set up tracing
+trace_provider = TracerProvider(resource=resource)
+trace_provider.add_span_processor(BatchSpanProcessor(ConsoleSpanExporter()))
+trace.set_tracer_provider(trace_provider)
+
+# Set up logging
+logger_provider = LoggerProvider(resource=resource)
+logger_provider.add_log_record_processor(BatchLogRecordProcessor(ConsoleLogExporter()))
+set_logger_provider(logger_provider)
+
+# Attach OpenTelemetry to FastMCP's logger
+fastmcp_logger = get_logger("my_server")
+fastmcp_logger.addHandler(LoggingHandler(logger_provider=logger_provider))
+
+# Create your FastMCP server
+mcp = FastMCP("My Server")
+
+@mcp.tool()
+def greet(name: str) -> str:
+    """Greet someone by name."""
+    fastmcp_logger.info(f"Greeting {name}")
+    return f"Hello, {name}!"
+```
+
+### Production OTLP Export
+
+For production environments, replace console exporters with OTLP exporters:
+
+```python
+from opentelemetry.exporter.otlp.proto.grpc.trace_exporter import OTLPSpanExporter
+from opentelemetry.exporter.otlp.proto.grpc._log_exporter import OTLPLogExporter
+
+# Configure OTLP endpoint (e.g., Grafana, Jaeger, or any OTLP collector)
+otlp_endpoint = "http://localhost:4317"
+
+# Tracing
+trace_provider = TracerProvider(resource=resource)
+trace_provider.add_span_processor(
+    BatchSpanProcessor(OTLPSpanExporter(endpoint=otlp_endpoint))
+)
+trace.set_tracer_provider(trace_provider)
+
+# Logging
+logger_provider = LoggerProvider(resource=resource)
+logger_provider.add_log_record_processor(
+    BatchLogRecordProcessor(OTLPLogExporter(endpoint=otlp_endpoint))
+)
+set_logger_provider(logger_provider)
+```
+
+### Structured Logging with OpenTelemetry
+
+FastMCP's `StructuredLoggingMiddleware` outputs JSON logs that OpenTelemetry collectors can parse and enrich:
+
+```python
+from fastmcp import FastMCP
+from fastmcp.server.middleware.logging import StructuredLoggingMiddleware
+
+mcp = FastMCP("Structured Server")
+
+# Add structured logging middleware
+mcp.add_middleware(StructuredLoggingMiddleware(
+    include_payloads=True,
+    max_payload_length=1000
+))
+
+# OpenTelemetry will capture these structured logs
+```
+
+The structured logs include metadata like request timestamps, method names, token estimates, and payload sizes - perfect for observability platforms.
+
+## Spans via Middleware
+
+FastMCP's middleware system provides the perfect foundation for creating OpenTelemetry spans that track MCP operations.
+
+### Basic Tracing Middleware
+
+Create a middleware that emits spans for all MCP requests:
```

Comment:
Couldn't we just ship this middleware with FastMCP?  I'd argue it should be on by default, even.